csharp/src/Google.Protobuf.Test/ReadOnlySequenceFactory.cs

```diff
@@ -50,9 +50,9 @@ public static ReadOnlySequence<byte> CreateWithContent(byte[] data, int segmentS
             while (currentIndex < data.Length)
             {
                 var segment = new List<byte>();
-                for (; currentIndex < Math.Min(currentIndex + segmentSize, data.Length); currentIndex++)
+                while (segment.Count < segmentSize && currentIndex < data.Length)
```

Comment:
FYI there was a bug here that created a sequence that started and ended with empty segments, but the actual content was a single segment.

Reply:
Thanks for the fix!